collector-sqs/src/main/java/zipkin/collector/sqs/SQSSpanProcessor.java

```diff
@@ -112,9 +112,12 @@ private void process(final List<Message> messages) {
           }
 
           @Override public void onError(Throwable t) {
-            // don't delete messages. this will allow accept calls retry once the
-            // messages are marked visible by sqs.
             logger.log(Level.WARNING, "collector accept failed", t);
+            // for cases that are not recoverable just discard the message,
+            // otherwise ignore so processing can be retried.
+            if (t instanceof IllegalArgumentException && t.getMessage().contains("Malformed")) {
```

Comment:
wonder if we should special case it to malformed message or just anything illegal? not sure we are 100% consistent

Reply:
I was pondering that question as well.  Its probably safer to remove the special case and throw out anything illegal.  Will update the PR.